test/test.h

```diff
@@ -115,9 +115,9 @@ class assert_stream {
 // Make a new test object. The body of the test should follow this
 // macro, e.g. TEST(equality) { ASSERT(1 == 1); }
 #define TEST(name)                                                     \
-  void test_##name##_body();                                           \
+  __device__ void test_##name##_body();                                           \
```

Comment:
Doesn't this also need to be `__host__`? Surely most of the tests don't work due to use of `array<>`?

Maybe just use `NDARRAY_HOST_DEVICE`, so it works on non-cuda builds too?

Reply:
I'm going to revert this - there's no (easy) way to run these tests in CI anyway. Let's just stick with a build test - for those I'll copy a subset of the existing tests and get this PR merged.